public/angular/js/controllers/tags/report_tags_modal.js

```diff
@@ -0,0 +1,250 @@
+          $scope.selectedReport.read = true;
+        }
+        $scope.saveReport();
+      };
+
+      $scope.isRead = function() {
+        return $scope.selectedReport.read;
+      }
+
+      // Tagify Event Listeners
+      $scope.onAddSMTCTag = function(e, addedTag) {
+
+      };
+
+      $scope.onRemoveSMTCTag = function(e, addedTag) {
+      };
+
+      $scope.onInvalidSMTCTagAdd = function(e, addedTag) {
+      };
+
+      $scope.removeAllTags = function() {
+        $scope.tagifyElement.removeAllTags();
+        $scope.addedSMTCTagsIds = [];
+      }
+
+      $scope.loadReportTags = function(tagifyElement) {
+        $scope.selectedReport.smtcTags.forEach( function(tag) {
+          tagifyElement.addTags([{
+            value: $scope.smtcTagsById[tag].name,
+            title: $scope.smtcTagsById[tag]._id
+          }]);
+        })
+      }
+
+      $scope.tagify = function() {
+        /* Because we are manipulating the DOM, tag inputs don't actually load on init(). This means that running a
+         * forEach statement in init (like linkify) doesn't work because DOM elements can't be found. Instead this
+         * function is run when the element is loaded using ng-init. THIS IS NOT HOW NG-INIT is supposed to be
+         * used. I would otherwise use a component, but this was coded in angular 1.2 which doesn't have good component
+         * support.
+         */
+        var tagifyElement = angular.element('input[name=tagifySMTCTags]');
+        // Add Translated Placeholder Text
+        tagifyElement.attr("placeholder", $scope.tagifyPlaceHolderText);
+        tagifyElement.tagify({
+            enforceWhitelist: true,
+            whitelist: tagifyFormatTags($scope.smtcTags),
+            autocomplete: true,
+            transformTag: transformTag,
+            dropdown : {
+              classname: "color-blue",
+              enabled: 0,              // show the dropdown immediately on focus
+              maxItems: 5,
+              position: "text",         // place the dropdown near the typed text
+              closeOnSelect: false,     // keep the dropdown open after selecting a suggestion
+              highlightFirst: true,
+            }
+        }).on('add', function(e, tagName){ // Attach Event Listeners
+            $scope.onAddSMTCTag(e, tagName);
+          })
+          .on('removeTag', function(e, tagName){
+            $scope.onRemoveSMTCTag(e, tagName);
+          })
+          .on("invalid", function(e, tagName) {
+            $scope.onInvalidSMTCTagAdd(e, tagName);
+          });
+        $scope.tagifyElement = tagifyElement.data('tagify');
+        $scope.loadReportTags($scope.tagifyElement);
+      };
+
+      $scope.sourceClass = function() {
+        // Pick one of the sources that has a media type. For now, it happens that
+        // if a report has multiple sources, they all have the same type, or are
+        // deleted
+        if ($scope.selectedReport.metadata.platform === "Facebook") {
```

Comment:
I was thinking we should make some constants.js(or something similar) file that contains all these types of Constant String values. Because I also have been hardcoding strings which I dont think is a good approach. I will try to add that in next PR.

Reply:
Sounds good. Please do.